### PR TITLE
status: detect when DaemonSets and Deployments are rolling out

### PR DESCRIPTION
**Pull request by squeed**
There are a few additional states that indicate a Deployment or DaemonSet is progressing and not yet fully deployed:

- ObservedGeneration < Generation means the relevant controller has not yet observed the changes
- UpdatedReplicas / UpdatedNumberScheduled too small means that the update has not fully rolled out.

  Fixes: #413 which was somehow reverted in a refactor

Signed-off-by: Casey Callendrello <cdc@isovalent.com>

### CODE REVIEW

File: status/k8s.go [hunk 1]
```diff
@@ -12,14 +12,15 @@ import (
 	"sync"
 	"time"
 
-	"github.com/cilium/cilium/api/v1/models"
-	ciliumv2 "github.com/cilium/cilium/pkg/k8s/apis/cilium.io/v2"
```
Comment (squeed) [inline status/k8s.go:16]: oops, gofmt misbehaved. I'll fix after review.

**Review by gandro — APPROVED** [review]
Thanks!

**Comment by squeed** [comment]
Test failure is a flake connecting to google.com. Definitely not related to this PR.

re-pushed anyways to address goimport issue.

**Review by ldelossa — APPROVED** [review]
very handy!

**Review by tklauser — APPROVED** [review]

**Pull request merged**